src/interface/pipe.rs

```diff
@@ -142,6 +142,29 @@ impl EmulatedPipe {
         bytes_written as i32
     }
 
+    // Write length bytes from pointer into pipe
+    pub fn write_vectored_to_pipe(
+        &self,
+        ptr: *const interface::IovecStruct,
+        iovcnt: i32,
+        nonblocking: bool,
+    ) -> i32 {
+        let mut buf = Vec::new();
+        let mut length = 0;
+
+        for _iov in 0..iovcnt {
+            unsafe {
+                let iovec = *ptr;
+                assert!(!ptr.is_null());
```

Comment:
This check is performed after dereferencing the pointer, which is not right to me. This assertion should be placed before dereferencing to prevent undefined behavior.